sqlg-core/src/main/java/org/umlg/sqlg/structure/SqlgDataSourceFactory.java

```diff
@@ -0,0 +1,20 @@
+package org.umlg.sqlg.structure;
+
+import org.apache.commons.configuration.Configuration;
+import javax.sql.DataSource;
+
+/**
+ * Created by petercipov on 27/02/2017.
+ */
+public interface SqlgDataSourceFactory {
+
+    SqlgDataSource setup(String driver, final Configuration configuration) throws Exception;
+    SqlgDataSource setupFromJndi(String jndiName, Configuration configuration) throws Exception;
```

Comment:
This is IMHO not the right place to handle JNDI datasources. When you obtain a datasource from JNDI, it is already bound to some URL and all you can do is to get an actual connection from the datasource. I.e. `SqlgDataSource#get(String)` method does not make sense for JNDI-based connections.